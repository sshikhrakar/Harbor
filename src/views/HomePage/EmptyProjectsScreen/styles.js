import { StyleSheet } from 'react-native';

import fonts from '../../../config/fonts';
import colors from '../../../config/colors';

const styles = StyleSheet.create({
  emptyProjectsContainer: {
    flex: 1,
    backgroundColor: colors.SILVER,
  },

  imageContainer: {
    flex: 1,
    padding: 20,
    alignItems: 'center',
    justifyContent: 'flex-end',
  },

  image: {
    flex: 0.3,
    resizeMode: 'contain',
  },

  textContainer: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'flex-start',
  },

  mainTextContainer: {
    marginVertical: 8,
  },

  mainText: {
    fontSize: 24,
    textAlign: 'center',
    fontFamily: fonts.primary.REGULAR,
  },

  subText: {
    fontSize: 16,
    fontFamily: fonts.primary.LIGHT,
  },
});

export default styles;
