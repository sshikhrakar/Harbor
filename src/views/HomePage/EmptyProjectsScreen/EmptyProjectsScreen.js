import React from 'react';
import {
  View,
  Text,
  Image,
} from 'react-native';

import styles from './styles';
import images from '../../../config/images';

function EmptyProjectsScreen() {

  return (
    <View style={ styles.emptyProjectsContainer }>
      <View style={ styles.imageContainer }>
        <Image
          source={ images.indifferentEmoji }
          style={ styles.image }
        />
      </View>

      <View style={ styles.textContainer }>
        <View style={ styles.mainTextContainer }>
          <Text style={ styles.mainText }>
            { `You don't seem to have added any projects..` }
          </Text>
        </View>

        <View style={ styles.subTextContainer }>
          <Text style={ styles.subText }>
            { `Press the Add button on the top right to get started` }
          </Text>
        </View>
      </View>
    </View>
  );

}

export default EmptyProjectsScreen;
