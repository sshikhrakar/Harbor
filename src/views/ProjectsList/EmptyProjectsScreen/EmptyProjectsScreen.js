import React from 'react';
import PropTypes from 'prop-types';
import {
  View,
  Text,
  Image,
  ScrollView,
  RefreshControl,
} from 'react-native';

import styles from './styles';
import fonts from '../../../config/fonts';
import colors from '../../../config/colors';
import images from '../../../config/images';

function EmptyProjectsScreen(props) {
  const {
    isFetching,
    fetchAllProjects,
  } = props;

  return (
    <ScrollView
      refreshControl={
        <RefreshControl
          refreshing={ isFetching }
          onRefresh={ fetchAllProjects }
        />
      }
      contentContainerStyle={ styles.scrollViewContainer }
    >
      <View style={ styles.emptyProjectsContainer }>
        <View style={ styles.imageContainer }>
          <Image
            source={ images.indifferentEmoji }
            style={ styles.image }
          />
        </View>

        <View style={ styles.textContainer }>
          <View style={ styles.mainTextContainer }>
            <Text style={ styles.mainText }>
              { `You don't seem to have been invited to any projects ...` }
            </Text>
          </View>

        </View>
      </View>
    </ScrollView>
  );

}

EmptyProjectsScreen.propTypes = {
  isFetching: PropTypes.bool.isRequired,
  fetchAllProjects: PropTypes.func.isRequired,
};

/*
 * Navigation bar style parameters.
 */
EmptyProjectsScreen.navigatorStyle = {
  navBarTextFontSize: 20,
  navBarTextFontFamily: fonts.primary.REGULAR,
  navBarComponentAlignment: 'center',
  navBarBackgroundColor: colors.SILVER,
};


export default EmptyProjectsScreen;
