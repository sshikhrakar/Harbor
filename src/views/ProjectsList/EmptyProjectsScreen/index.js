import EmptyProjectsScreen from './EmptyProjectsScreen';

export default EmptyProjectsScreen;
