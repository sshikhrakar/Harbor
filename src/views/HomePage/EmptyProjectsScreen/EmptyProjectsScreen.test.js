import React from 'react';
import renderer from 'react-test-renderer';

import EmptyProjectsScreen from './EmptyProjectsScreen';

describe('EmptyProjectsScreen View', () => {

  it('should render correctly', () => {
    const tree = renderer.create(
      <EmptyProjectsScreen />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

});
