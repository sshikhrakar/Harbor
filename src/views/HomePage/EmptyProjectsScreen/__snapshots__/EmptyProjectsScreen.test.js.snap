// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyProjectsScreen View should render correctly 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
  >
     HomePage 
  </Text>
</View>
`;
